test/unit/test_tpm2_session.c

```diff
@@ -0,0 +1,249 @@
+
+    TPM2B_ENCRYPTED_SECRET encrypted_salt;
+    memset(&encrypted_salt, 0, sizeof(encrypted_salt));
+
+    TPMT_SYM_DEF symmetric;
+    memset(&symmetric, 0, sizeof(symmetric));
+    symmetric.algorithm = TPM2_ALG_NULL;
+
+    TPM2B_NONCE nonce_caller;
+    memset(&nonce_caller, 0, sizeof(nonce_caller));
+    nonce_caller.size = tpm2_alg_util_get_hash_size(TPM2_ALG_SHA1);
+
+    set_expected(
+    TPM2_RH_NULL,
+    TPM2_RH_NULL, &encrypted_salt, session_type, &symmetric,
+    TPM2_ALG_SHA256, &nonce_caller, handle, rc);
+}
+
+TSS2_RC __wrap_Tss2_Sys_StartAuthSession(TSS2_SYS_CONTEXT *sysContext,
+        TPMI_DH_OBJECT tpmKey, TPMI_DH_ENTITY bind,
+        TSS2L_SYS_AUTH_COMMAND const *cmdAuthsArray,
+        const TPM2B_NONCE *nonceCaller,
+        const TPM2B_ENCRYPTED_SECRET *encryptedSalt, TPM2_SE sessionType,
+        const TPMT_SYM_DEF *symmetric, TPMI_ALG_HASH authHash,
+        TPMI_SH_AUTH_SESSION *sessionHandle, TPM2B_NONCE *nonceTPM,
+        TSS2L_SYS_AUTH_RESPONSE *rspAuthsArray) {
+
+    UNUSED(sysContext);
+    UNUSED(cmdAuthsArray);
+    UNUSED(nonceTPM);
+    UNUSED(rspAuthsArray);
+
+    expected_data *e = (expected_data *) mock();
+
+    assert_int_equal(tpmKey, e->input.key);
+
+    assert_int_equal(bind, e->input.bind);
+
+    assert_memory_equal(nonceCaller, &e->input.nonce_caller,
+            sizeof(*nonceCaller));
+
+    assert_memory_equal(encryptedSalt, &e->input.encrypted_salt,
+            sizeof(*encryptedSalt));
+
+    assert_int_equal(sessionType, e->input.session_type);
+
+    assert_memory_equal(symmetric, &e->input.symmetric, sizeof(*symmetric));
+
+    assert_int_equal(authHash, e->input.auth_hash);
+
+    *sessionHandle = e->output.handle;
+
+    TSS2_RC rc = e->output.rc;
+    free(e);
+    return rc;
+}
+
+#define TPM2B_PUBLIC_INIT(value) { \
+    .publicArea = { \
+        .objectAttributes = value \
+    } \
+}
+
+#define SAPI_CONTEXT   ((TSS2_SYS_CONTEXT *)0xDEADBEEF)
+#define SESSION_HANDLE 0xBADC0DE
+
+static void test_tpm2_session_defaults_good(void **state) {
+    UNUSED(state);
+
+    set_expected_defaults(TPM2_SE_POLICY, SESSION_HANDLE, TPM2_RC_SUCCESS);
+
+    tpm2_session_data *d = tpm2_session_data_new(TPM2_SE_POLICY);
+    assert_non_null(d);
+
+    tpm2_session *s = tpm2_session_new(SAPI_CONTEXT, d);
+    assert_non_null(s);
+
+    TPMI_SH_AUTH_SESSION handle = tpm2_session_get_session_handle(s);
+    assert_int_equal(handle, SESSION_HANDLE);
+
+    TPMI_ALG_HASH auth_hash = tpm2_session_get_authhash(s);
+    assert_int_equal(auth_hash, TPM2_ALG_SHA256);
+
+    tpm2_session_free(&s);
+    assert_null(s);
+}
+
+static void test_tpm2_session_setters_good(void **state) {
+    UNUSED(state);
+
+    tpm2_session_data *d = tpm2_session_data_new(TPM2_SE_TRIAL);
+    assert_non_null(d);
+
+    tpm2_session_set_authhash(d, TPM2_ALG_SHA512);
+
+    TPMT_SYM_DEF symmetric = { .algorithm = TPM2_ALG_AES, .keyBits = { .aes =
+            256 }, .mode = { .aes = 42 }, };
```

Comment:
The format here is a bit off too. It makes really hard to read.

Reply:
This isn't he formatter yet again.

Reply:
This *is* the formatter